01-node-tutorial/answers/customEmitter.js

```diff
@@ -0,0 +1,19 @@
+const EventEmmiter = require("events");
+
+const emitter = new EventEmmiter();
+// setInterval(() => {
+//     emitter.emit("timer", "hi there");
+// }, 2000);
+// emitter.on("timer", (msg) => console.log(msg));
+
+const waitForEvent = () => {
+    return new Promise((resolve) => {
+        emitter.on("happens", (msg) => resolve(msg));
+    })
+};
+const doWait = async () => {
+    const msg = await waitForEvent();
+    console.log("We got an event! it is: ", msg);
```

Comment:
great job, however, you may want to write code that is different from the sample used in the assignment so you can be more creative